src/adera/Machines/Rocket.cpp

```diff
@@ -92,7 +101,40 @@ void SysMachineRocket::update_physics(ActiveScene& rScene)
             continue;
         }
 
-        //std::cout << "updating a rocket\n";
+        // Check for nonzero throttle, continue otherwise
+        WireData *throttle = machine.m_wiThrottle.connected_value();
+        wiretype::Percent* percent = nullptr;
+        if (throttle)
+        {
+            using wiretype::Percent;
+            percent = std::get_if<Percent>(throttle);
+            if (!(percent->m_value > 0.0f))
+            {
+                continue;
+            }
+        }
+        else { continue; }
+
+        // Check for adequate resource inputs
+        bool fail = false;
+        for (auto& resource : machine.m_resourceLines)
+        {
+            MachineContainer* src = m_scene.reg_try_get<MachineContainer>(resource.m_sourceEnt);
```

Comment:
I think ultimately some of the systems are going to get pretty complicated. Rocket engines in particular *are* kind of complicated because you have pretty much every possible system either driving them or being driven by them. The rocket engine depends on user command inputs, fuel availability, more complicated physics simulation stuff like temperature and fuel flow rate, and drives other physics (ship rigidbody physics, thermal stuff), fuel depletion, and graphical effects. I vote we keep expanding it until it becomes obvious that something should be separated. When we have a more functional game I'm down to try to optimize the architecture more, but frankly my goal at the moment is to start the "bloat it with features" stage of development and let that drive engine development and architecture discussions when we notice something getting ugly.

> "To properly plan a project, you have to do the project"
>  -- Capital-Asterisk [UTC-7]é#@ñ±, 11/17/2020